js/packages/teams-ai/src/Authentication.ts

```diff
@@ -151,15 +159,37 @@ export class Authentication<TState extends TurnState = DefaultTurnState> {
     /**
      * @private
      */
-    private async runDialog(context: TurnContext, state: TState, dialogStateProperty: string): Promise<DialogTurnResult<OAuthPromptResult>> {
+    private async runDialog(context: TurnContext, state: TState, dialogStateProperty: string): Promise<DialogTurnResult<any>> {
         // Save the
         const accessor = new TurnStateProperty<DialogState>(state, 'conversation', dialogStateProperty);
         const dialogSet = new DialogSet(accessor);
-        dialogSet.add(this._oauthPrompt);
+        dialogSet.add(this._prompt);
+        dialogSet.add(new WaterfallDialog('taskNeedingAuth', [
+          async (step) => {
+            return await step.beginDialog(promptName);
+          },
+          async (step) => {
+            const token = step.result;
```

Comment:
to me it's a little confusing that this is called `token`. Any chance of using a different name so that `<somethingElse>.token` is used below?